app/Http/Controllers/MerchandiseController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+// phpcs:disable SlevomatCodingStandard.Functions.DisallowNamedArguments
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\DuesTransactionMerchandise as DuesTransactionMerchandiseResource;
+use App\Http\Resources\Merchandise as MerchandiseResource;
+use App\Http\Resources\User as UserResource;
+use App\Models\DuesTransactionMerchandise;
+use App\Models\Merchandise;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+class MerchandiseController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('permission:read-merchandise|distribute-swag', ['only' => ['index']]);
+        $this->middleware('permission:distribute-swag', ['only' => ['getDistribution', 'distribute']]);
+    }
+
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(): JsonResponse
+    {
+        return response()->json(
+            [
+                'status' => 'success',
+                'merchandise' => MerchandiseResource::collection(
+                    Merchandise::whereDistributable(true)
+                        ->orderByDesc('fiscal_year_id')
+                        ->get()
+                ),
+            ]
+        );
+    }
+
+    public function getDistribution(Merchandise $merchandise, User $user): JsonResponse
+    {
+        if (! $merchandise->distributable) {
+            return response()->json(
+                data: [
+                    'status' => 'error',
+                    'message' => 'item is not distributable',
```

Comment:
Can we make these sentence case and presentable so we can just directly show the error message in the UI?

like here `ITEM_NOT_DISTIRBUTABLE` and the app can map on its own, or the error message from the backend should be suitable to drop in the UI